Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -325,11 +325,34 @@ def check_binary_arches(formula)
     end
     return if mismatches.empty?
 
-    <<~EOS
-      Binaries built for a non-native architecture were installed into #{formula}'s prefix.
-      The offending files are:
-        #{mismatches * "\n        "}
-    EOS
+    compatible_universal_binaries = mismatches.select do |file|
+      file.arch == :universal && file.archs.include?(Hardware::CPU.arch)
+    end
+    mismatches -= compatible_universal_binaries
+
+    universal_binaries_expected =
+      formula.tap.present? && tap_audit_exception(:universal_binary_allowlist, formula.name)
```

Comment:
Alternatively, we could also just enforce this for Homebrew/core with
```suggestion
    universal_binaries_expected = if formula.tap.present?
      formula.tap.core_tap? && tap_audit_exception(:universal_binary_allowlist, formula.name)
    else
      true
    end
```

Reply:
Yeh, I'd be tempted to make this core-only.

Reply:
Check should be core-only now. Checking against the allowlist is also silently skipped if you don't have Homebrew/core tapped, which I think might be what we want.